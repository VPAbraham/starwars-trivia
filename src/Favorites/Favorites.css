.section--favorites {
  font-family: 'Nanum Gothic Coding', monospace;
  /* font-family: 'Ubuntu Mono', monospace; */

}

.header--favorites {
  font-size: 2em;
}

.div--favorites {
  
}