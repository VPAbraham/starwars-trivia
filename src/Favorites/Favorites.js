import React from 'react';
import CharacterCard from '../CharacterCard/CharacterCard';
import './Favorites.css';
import propTypes from 'prop-types'



const Favorites = (props) => {
  const favoriteCards = props.favorites.length ? props.favorites.map(favorite => {
    return ((<li key={favorite.name} className='li li--character'>
      <CharacterCard
        name={favorite.name}
        homeworld={favorite.homeworld}
        species={favorite.species}
        relatedFilms={favorite.relatedFilms}
        isFavorite={false}
        method={props.removeFav}
      />
    </li>) 
    )
  }) : <p>You have no favorite characters yet!</p>
  return (
    <section className='section--character-list'>
      <header className='header--favorites'>Favorites</header>
      <ul className='ul ul--character-list'>
        {favoriteCards}
      </ul>
    </section>
  )
}

export default Favorites;

Favorites.propTypes = {
  favorites: propTypes.array
}