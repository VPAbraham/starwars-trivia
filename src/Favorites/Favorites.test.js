import React from 'react';
import { shallow } from 'enzyme';
import Favorites from './Favorites'


describe('Favorites', () => {
  let wrapper, favProps;
  beforeEach(() => {
    favProps = {
      favorites: []
    }

    wrapper = shallow(<Favorites favProps = {favProps} />)
  });
  it('should match snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  });
});