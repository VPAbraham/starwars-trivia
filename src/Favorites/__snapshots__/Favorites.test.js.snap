// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot 1`] = `
<section
  className="section--character-list"
>
  <header
    className="header--favorites"
  >
    Favorites
  </header>
  <ul
    className="ul ul--character-list"
  >
    <li
      className="li li--character"
      key="char1"
    >
      <CharacterCard
        isFavorite={false}
        name="char1"
      />
    </li>
  </ul>
</section>
`;
